### PR TITLE
actions: Replace in-repo actions with separate repo

### PR DESCRIPTION
**Pull request by teodutu**
Storing GitHub actions inside the repo creates unnecessary duplicates across multiple repos. This issue is fixed by referring a centralised repo that stores actions as `.yml` files and basic configs.

### CODE REVIEW

**Comment by teodutu** [comment]
@gabrielmocanu Running the labeller results in some 403s [1]. Should I keep some config files locally or something?
[1] https://github.com/open-education-hub/operating-systems/actions/runs/5698005800/job/15445395870?pr=315

File: .github/workflows/labeler.yml [hunk 1]
```diff
@@ -1,26 +1,36 @@
...
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions/github-script@v6
-        with:
-          script: |
-            const { default: assignLabelsToPRs } = await import('${{ github.workspace }}/.github/scripts/add-labels.js') ;
-            await assignLabelsToPRs({github, context})
+topic/software-stack:
+  - content/chapters/software-stack/**/*
```
Comment (gabrielmocanu) [inline .github/workflows/labeler.yml:2]:
```suggestion
  - 'content/chapters/software-stack/**/*'
```

File: .github/workflows/labeler.yml [hunk 2]
```diff
@@ -1,26 +1,36 @@
...
+
+topic/data:
+  - content/chapters/data/**/*
```
Comment (gabrielmocanu) [inline .github/workflows/labeler.yml:5]:
```suggestion
  - 'content/chapters/data/**/*'
```

File: .github/workflows/labeler.yml [hunk 3]
```diff
@@ -1,26 +1,36 @@
...
+
+topic/compute:
+  - content/chapters/compute/**/*
```
Comment (gabrielmocanu) [inline .github/workflows/labeler.yml:8]:
```suggestion
  - 'content/chapters/compute/**/*'
```

File: .github/workflows/labeler.yml [hunk 4]
```diff
@@ -1,26 +1,36 @@
...
+
+topic/io:
+  - content/chapters/io/**/*
```
Comment (gabrielmocanu) [inline .github/workflows/labeler.yml:11]:
```suggestion
  - 'content/chapters/io/**/*'
```

File: .github/workflows/labeler.yml [hunk 5]
```diff
@@ -1,26 +1,36 @@
...
+
+topic/app-interact:
+  - content/chapters/app-interact/**/*
```
Comment (gabrielmocanu) [inline .github/workflows/labeler.yml:14]:
```suggestion
  - 'content/chapters/app-interact/**/*'
```

File: .github/workflows/labeler.yml [hunk 6]
```diff
@@ -1,26 +1,36 @@
...
+  - '**/lab/solution/**/*'
+  - '**/lecture/demo/**/*'
+
+area/infra:
+  - 'util/**/*'
+  - '.github/**/*'
+  - 'Dockerfile'
+
+area/assignment:
+  - content/assignments/**/*
```
Comment (gabrielmocanu) [inline .github/workflows/labeler.yml:36]:
```suggestion
  - 'content/assignments/**/*'
```

**Review by gabrielmocanu — CHANGES_REQUESTED** [review]

File: .github/workflows/labeler.yml (hunk 1 above)
Comment (teodutu) [reply .github/workflows/labeler.yml:2]: Why though? It was working just fine before.

**Review by gabrielmocanu — APPROVED** [review]

File: .github/workflows/labeler.yml (hunk 3 above)
Comment (PAN740623) [reply .github/workflows/labeler.yml]: 更改

File: .github/workflows/labeler.yml (hunk 4 above)
Comment (PAN740623) [reply .github/workflows/labeler.yml]: 更改

File: .github/workflows/labeler.yml (hunk 5 above)
Comment (PAN740623) [reply .github/workflows/labeler.yml]: 更改

File: .github/workflows/labeler.yml (hunk 6 above)
Comment (PAN740623) [reply .github/workflows/labeler.yml]: 更改

File: .github/workflows/labeler.yml (hunk 2 above)
Comment (PAN740623) [reply .github/workflows/labeler.yml]: 更改

File: .github/workflows/labeler.yml (hunk 1 above)
Comment (PAN740623) [reply .github/workflows/labeler.yml]: 更改

**Pull request merged**